clients/githubrepo/contributors.go

```diff
@@ -42,50 +56,170 @@ func (handler *contributorsHandler) init(ctx context.Context, repourl *Repo) {
-			if err == nil {
-				for _, org := range orgs {
-					contributor.Organizations = append(contributor.Organizations, clients.User{
-						Login: org.GetLogin(),
-					})
-				}
+			if users != nil {
+				owners = append(owners, users...)
 			}
-			user, _, err := handler.ghClient.Users.Get(handler.ctx, contrib.GetLogin())
-			if err != nil {
-				handler.errSetup = fmt.Errorf("error during Users.Get: %w", err)
+		}
+	}
+
+	var ownerUsers []clients.User
+	for _, own := range owners {
+		if own.GetLogin() == "" {
+			continue
+		}
+		owner := clients.User{
+			Login:       own.GetLogin(),
+			IsCodeOwner: true,
+		}
+
+		// if verified external contributor add repo org to organization list by default
+		if ok := slices.Contains(verifiedExternalOwners, owner.Login); ok {
+			owner.Organizations = append(owner.Organizations, clients.User{
+				Login: handler.repourl.owner,
+			})
+		}
+
+		orgs, _, err := handler.ghClient.Organizations.List(handler.ctx, own.GetLogin(), nil)
+		// This call can fail due to token scopes. So ignore error.
+		if err == nil {
+			for _, org := range orgs {
+				owner.Organizations = append(owner.Organizations, clients.User{
+					Login: org.GetLogin(),
+				})
 			}
-			contributor.Companies = append(contributor.Companies, user.GetCompany())
-			handler.contributors = append(handler.contributors, contributor)
 		}
-		handler.errSetup = nil
-	})
-	return handler.errSetup
+		owner.Companies = append(owner.Companies, own.GetCompany())
+
+		ownerUsers = append(ownerUsers, owner)
+	}
+	return ownerUsers
+}
+
+// expand team owners to multiple github users.
+func getUsersFromTeam(handler *contributorsHandler, owner codeowners.Owner) []*github.User {
+	users, response, err := handler.ghClient.Teams.ListTeamMembersBySlug(
+		handler.ctx,
+		handler.repourl.owner,
+		owner.Value,
+		&github.TeamListTeamMembersOptions{},
+	)
+	if err == nil && response.StatusCode == http.StatusOK {
+		return users
+	}
+	return nil
+}
+
+// get github user from owner username.
+func getUserFromUsername(handler *contributorsHandler, owner codeowners.Owner) []*github.User {
+	user, response, err := handler.ghClient.Users.Get(handler.ctx, owner.Value)
+	if err == nil && response.StatusCode == http.StatusOK {
+		return []*github.User{user}
+	}
+	return nil
+}
+
+// get github user from email.
+func getUserFromEmail(handler *contributorsHandler, owner codeowners.Owner) []*github.User {
+	query := fmt.Sprintf("\"%s\" in:email", owner.String())
+	userSearchResults, response, err := handler.ghClient.Search.Users(handler.ctx, query, &github.SearchOptions{})
+	if err == nil && response.StatusCode == http.StatusOK && *userSearchResults.Total > 0 {
+		return []*github.User{userSearchResults.Users[0]}
+	}
+	return nil
+}
+
+// getting verified external owners by @verified comment.
+func getVerifiedExternalOwners(fileReader io.ReadCloser) []string {
+	verifiedExternalOwners := make([]string, 0)
+	r := regexp.MustCompile("^# @verified .*")
```

Comment:
remind me, is this verified comment something that has external semantics? Or what the purpose of it was?

Reply:
I need to check the original comment, it's explained there.